src/quote/leveraged-extended/utils/issuance-call.ts

```diff
@@ -0,0 +1,81 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { bytesToBigInt, fromHex, Hex } from 'viem'
+
+import { Exchange, getFlashMintLeveragedContractForToken } from 'utils'
+
+import { FlashMintLeveragedExtendedQuoteRequest } from '../provider'
+
+const ADDRESS_ZERO = '0x0000000000000000000000000000000000000000'
+const chainId = 42161
+
+export async function getIndexTokenAmount(
+  request: FlashMintLeveragedExtendedQuoteRequest,
+  provider: JsonRpcProvider
+): Promise<BigNumber> {
+  const { inputToken, inputTokenAmount, outputToken } = request
+  const indexToken = outputToken.address
+  const isInputTokenEth = request.inputToken.symbol === 'ETH'
+  // TODO: can this be zero?
+  const minIndexTokenAmount = BigInt('1000000000000')
```

Comment:
Current obstacle: how to determine a min amount here?

What I tried so far:
* setting to `0` doesn't work as Uniswap will return `AS: amountSpecified cannot be zero` ([error codes](https://docs.uniswap.org/contracts/v3/reference/error-codes))
* setting an arbitrary small amount doesn't work either as then the call will revert with `Invalid transfer in. Results in undercollateralization`

Then in general the question for the whole approach of trying to static call the function (as you mentioned before) is, what if the user has not approved tokens (when using ERC-20 as input token) or insufficient funds.

Reply:
Side note: `Invalid transfer in. Results in undercollateralization` is the bug that will be fixed with the new audited contract.